Add a ‘remove end gutters’ option

johngodley opened the pull request:
This adds an option to remove the end gutters from the layout grid. The option is available in the parent grid:

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75871059-e352f900-5e03-11ea-8596-19767f6e3440.jpg)

Enabling the option removes gutters in the editor.

Before:

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75871152-05e51200-5e04-11ea-801d-0fe7b6d541e2.jpg)

After:

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75871220-2319e080-5e04-11ea-8591-c436ff88c2cc.jpg)

The same applies to the front end.

Fixes #29

johngodley commented:
@jasmussen appreciate any feedback here. I've gone for a negative option (i.e. you enable the option to disable something) due to existing blocks not having the feature, but the logic could be inverted if that's better.

jasmussen commented:
Here's what I'm seeing:

![gutters](https://user-images.githubusercontent.com/1204802/75872603-c53ec600-5e0e-11ea-9974-5128fc065d9d.gif)

This seems generally good, except for a few things.

First off the behavior — "Remove end gutters" off by default, or "Add end gutters" on by default? Right at this moment, I'm leaning towards the latter.

Let's go with a switch, btw, instead of a checkbox, as the visual effect is instant. So:

```
[switch] Add end gutters
Toggle to remove the spacing left and right of the grid.
```

I don't love the verbiage, but it's probably fine for a start.

There's an issue with the alignment, though, I think we need to remove the end gutters differently. It looks like the toggle correctly removes the end gutters from the actual _content_, but it doesn't remove the end gutters from the `.wpcom-overlay-grid`. Here's a snippet:

```
.wpcom-overlay-grid {
	// The following rules are for Safari.
	// A CSS grid rendering issue means the child nodes of this grid
	// won't stretch to fit the vertical space, unless the position
	// property is changed from absolute, to something else, and back.
	// By being relative when unselected and absolute when selected, we
	// constantly "refresh" this hack.
	position: relative;
	.is-hovered &,
	.is-selected &,
	.has-child-selected & {
		position: absolute;
	}

	top: 0;
	left: 0;
	bottom: 0;
	right: 0;
	display: grid;
	grid-gap: 24px;
	grid-template-columns: $grid-desktop;

	// This padding adds end gutters.
	padding-left: 24px;
	padding-right: 24px;
```

If you toggle out those two paddings at the end at the same time as for the content, this PR should be good to go!

johngodley commented:
> It looks like the toggle correctly removes the end gutters from the actual content, but it doesn't remove the end gutters from the .wpcom-overlay-grid

Yep, good point - missed that. Added

> Let's go with a switch

I've switched to `ToggleControl`, if that's what you mean:

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75887828-4b183c80-5e22-11ea-9895-b1db77e1ed74.jpg)

The logic is now inverted

jasmussen commented:
> The logic is now inverted

Looks good, but I feel like it should be enabled by default.

johngodley commented:
It is enabled by default. The screenshot is just whatever state the block was in when I took it

jasmussen commented:
Ship it!

jasmussen commented:
Hmm actually we may want conditional help text.

johngodley commented:
Alright, message also toggles:

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75901411-5d9c7100-5e36-11ea-8735-99e24bf82e27.jpg)

![Edit_Post_‹_Latest_—_WordPress](https://user-images.githubusercontent.com/1277682/75901438-6725d900-5e36-11ea-8e53-5bc4743b0dfd.jpg)

jasmussen commented:
Okay now ship it. And thank you!
